src/mem/cache/tags/tagged_entry.hh

```diff
@@ -31,44 +43,113 @@
 
 #include <cassert>
 
-#include "base/cache/cache_entry.hh"
 #include "base/cprintf.hh"
-#include "base/logging.hh"
 #include "base/types.hh"
 #include "mem/cache/replacement_policies/replaceable_entry.hh"
+#include "mem/cache/tags/indexing_policies/base.hh"
+#include "params/TaggedIndexingPolicy.hh"
+#include "params/TaggedSetAssociative.hh"
 
 namespace gem5
 {
 
+struct TaggedTypes
```

Comment:
I totally agree with you, but on the other hand SecureIndexingPolicy/SecureSetAssociative might sounds like something specific to the Secure address space (whereas it should handle Non-Secure address space as well)

Maybe more than Secure we should find a word which describes the presence of multiple address spaces?

Or maybe we could say that multiple address spaces are the norm, and we could rename



TaggedIndexingPolicy -> IndexingPolicy
TaggedSetAssociative -> SetAssociative

**Option1**:

BaseIndexingPolicy -> same
SetAssociative -> BaseSetAssociave (to match BaseIndexingPolicy)

**Option2**:

BaseIndexingPolicy -> GlobalIndexingPolicy
SetAssociative -> GlobalSetAssociative

Where the Global kw refers to the fact that it does not make distinction across address spaces...

What do you think?

Reply:
Out of the options you provided, I am leaning towards the Global option, but just weakly - it may work now, but if we end up having someone experimenting with more address space possibilities, this may introduce confusion.

I still think that although I recognize that the SecureAddr prefix is ambiguous, it should be ok to add it as a part of gem5´s learning curve, as long as we document it well enough to make it explicit that it can also contain addresses outside the secure space. I guess SecurableAddr resolves that ambiguity, but it may be a daunting name - nonetheless, with enough documentation it should be ok. Thoughts on SecurableAddr?

Reply:
Alright, you seem to have a better vision of what changes will be needed in the future to support those, so feel free to proceed naming it how you see fit :)